scib/metrics/isolated_labels.py

```diff
@@ -105,13 +105,18 @@ def isolated_labels_asw(
         )
 
     """
+    # precompute silhouette scores
+    adata.obs["silhouette_temp"] = silhouette_samples(
+        adata.obsm[embed], adata.obs[label_key]
+    )
     return isolated_labels(
         adata,
         label_key=label_key,
         batch_key=batch_key,
         embed=embed,
         cluster=False,
         iso_threshold=iso_threshold,
+        silhouette_pre=True,
```

Comment:
Would be good to have silhouette precomputation in the `score_isolated_label` function only